lib/hooks/useIsTouchdevice.ts

```diff
@@ -0,0 +1,15 @@
+import { useEffect, useState } from 'react'
+
+const useIsTouchdevice = (): boolean => {
+  const [isTouchdevice, setIsTouchdevice] = useState<boolean>()
+
+  useEffect(() => {
+    if (typeof window !== 'undefined') {
+      setIsTouchdevice(window.matchMedia('(hover: none)').matches)
```

Comment:
This seems like the right move now. Have you tried the newish `any-pointer`? 
I have not, so can't speak to it.